service/instance.py

```diff
@@ -268,11 +268,15 @@ def detach_port(esh_driver, esh_instance):
     return result
 
 
-def remove_network(esh_driver, identity_uuid, remove_network=False):
+def remove_network(esh_driver, identity_uuid):
```

Comment:
Remove `remove_empty_network` only does something if there are no active instances. By association `remove_network` only does something if there are no active instances (doesn't sit well with my intuition). 

provider actions like `stop_instance` Do the following:
```
    if reclaim_ip:
        remove_ips(esh_driver, esh_instance)
    stopped = esh_driver.stop_instance(esh_instance)
    if reclaim_ip:
        remove_network(esh_driver, identity_uuid)
```

It would be better like so:
```diff
    if reclaim_ip:
        remove_ips(esh_driver, esh_instance)
    stopped = esh_driver.stop_instance(esh_instance)
    if reclaim_ip:
-       remove_network(esh_driver, identity_uuid)
+      # Cleanup network if this is the last network
+      remove_empty_network(esh_driver, identity_uuid)
```

To be clear, i'm proposing renaming this to: `remove_empty_network`